pkg/asset/installconfig/powervs/session.go

```diff
@@ -182,6 +184,59 @@ func (c *BxClient) ValidateAccountPermissions() error {
 	return nil
 }
 
+//ValidateDhcpService checks for existing Dhcp service for the provided PowerVS cloud instance
+func (c *BxClient) ValidateDhcpService(ctx context.Context, svcInsID string) error {
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+
+	//create Power VS DHCP Client
+	dhcpClient := instance.NewIBMPIDhcpClient(ctx, c.PISession, svcInsID)
+	//Get all DHCP Services
+	dhcpServices, err := dhcpClient.GetAll()
+	if err != nil {
+		return errors.Wrap(err, "failed to get DHCP service details")
+	}
+	if len(dhcpServices) > 0 {
+		return fmt.Errorf("DHCP service already exists for provided cloud instance")
+	}
+	return nil
+}
+
+//ValidateCloudConnectionInPowerVSRegion counts cloud connection in PowerVS Region
+func (c *BxClient) ValidateCloudConnectionInPowerVSRegion(ctx context.Context, svcInsID string) error {
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	var cloudConnectionsIDs []string
+	cloudConnectionClient := instance.NewIBMPICloudConnectionClient(ctx, c.PISession, svcInsID)
+
+	//check number of cloudconnections
+	getAllResp, err := cloudConnectionClient.GetAll()
+	if err != nil {
+		return errors.Wrap(err, "failed to get existing Cloud connection details")
+	}
+	for _, cc := range getAllResp.CloudConnections {
+		cloudConnectionsIDs = append(cloudConnectionsIDs, *cc.CloudConnectionID)
+	}
+
+	//check for DHCP Service attached to Cloud connection
+	if len(cloudConnectionsIDs) >= 1 {
```

Comment:
can you omit this check? i could be wrong but i think that the range loop just won't do anything if there's nothing there, so you don't need this

Reply:
@clnperez As discussed I have removed the above check and added the check to verify number of cloud connection count per zone..